src/pip/_internal/pep425tags.py

```diff
@@ -419,11 +419,11 @@ def get_supported(
             supported.append(('%s%s' % (impl, versions[0]), abi, arch))
 
     # abi3 modules compatible with older version of Python
-    for version in versions[1:]:
-        # abi3 was introduced in Python 3.2
-        if version in {'31', '30'}:
-            break
-        for abi in abi3s:   # empty set if not Python 3
+    if supports_abi3:
+        for version in versions[1:]:
+            # abi3 was introduced in Python 3.2
+            if version in {'31', '30'}:
+                break
             for arch in arches:
                 supported.append(("%s%s" % (impl, version), abi, arch))
```

Comment:
`abi` seems now to be using the last value from the previous `for abi in abis:` loop a few lines above

Reply:
Thanks! Fixed [here](https://github.com/pypa/pip/pull/7367/files#diff-542f0dc2284dcb0cb6a0382dfeeb8ed2R427). I can't wait until we integrate the nice, thoroughly-tested, `packaging.tags`.